src/server/pfs/db/clock/clock.go

```diff
@@ -0,0 +1,153 @@
+package clock
+
+import (
+	"fmt"
+
+	"github.com/pachyderm/pachyderm/src/server/pfs/db/persist"
+)
+
+type ErrBranchExists struct {
+	error
+}
+
+type ErrBranchNotFound struct {
+	error
+}
+
+// NewClock returns a new clock for a given branch
+func NewClock(branch string) *persist.Clock {
+	return &persist.Clock{branch, 0}
+}
+
+// NewBranchClocks creates a new BranchClocks given a branch name
+// "master" -> [[(master, 0)]]
+func NewBranchClocks(branch string) persist.BranchClocks {
+	return persist.BranchClocks{{
+		Clocks: []*persist.Clock{{
+			Branch: branch,
+			Clock:  0,
+		}},
+	}}
+}
+
+// NewChild returns the child of a BranchClock
+// [(master, 0), (foo, 0)] -> [(master, 0), (foo, 1)]
+func NewChild(parent *persist.BranchClock) *persist.BranchClock {
+	if len(parent.Clocks) == 0 {
+		return parent
```

Comment:
Is this actually used somewhere? It seems kinda weird to define an empty clocks as being its own child.